xline-client/src/lib.rs

```diff
@@ -0,0 +1,399 @@
+    pub fn lease_client(&self) -> LeaseClient {
+        self.lease.clone()
+    }
+
+    /// Gets a lock client.
+    #[inline]
+    #[must_use]
+    pub fn lock_client(&self) -> LockClient {
+        self.lock.clone()
+    }
+
+    /// Gets a auth client.
+    #[inline]
+    #[must_use]
+    pub fn auth_client(&self) -> AuthClient {
+        self.auth.clone()
+    }
+
+    /// Gets a watch client.
+    #[inline]
+    #[must_use]
+    pub fn watch_client(&self) -> WatchClient {
+        self.watch.clone()
+    }
+
+    /// Gets a maintenance client.
+    #[inline]
+    #[must_use]
+    pub fn maintenance_client(&self) -> MaintenanceClient {
+        self.maintenance.clone()
+    }
+
+    /// Gets a cluster client.
+    #[inline]
+    #[must_use]
+    pub fn cluster_client(&self) -> ClusterClient {
+        self.cluster.clone()
+    }
+
+    /// Gets a election client.
+    #[inline]
+    #[must_use]
+    pub fn election_client(&self) -> ElectionClient {
+        self.election.clone()
+    }
+}
+
+/// Options for a client connection
+#[derive(Clone, Debug, Default)]
+pub struct ClientOptions {
+    /// User is a pair values of name and password
+    user: Option<(String, String)>,
+    /// Timeout settings for the curp client
+    curp_timeout: ClientTimeout,
+}
+
+impl ClientOptions {
+    /// Create a new `ClientOptions`
+    #[inline]
+    #[must_use]
+    pub fn new(user: Option<(String, String)>, curp_timeout: ClientTimeout) -> Self {
+        Self { user, curp_timeout }
+    }
+
+    /// Get `user`
+    #[inline]
+    #[must_use]
+    pub fn user(&self) -> Option<(String, String)> {
+        self.user.clone()
+    }
+
+    /// Get `curp_timeout`
+    #[inline]
+    #[must_use]
+    pub fn curp_timeout(&self) -> ClientTimeout {
+        self.curp_timeout
+    }
+}
+
+/// Authentication service.
+#[derive(Debug, Clone)]
+pub struct AuthService<S> {
+    /// A `Service` trait object
+    inner: S,
+    /// Auth token
+    token: Option<Arc<HeaderValue>>,
+}
+
+impl<S> AuthService<S> {
+    /// Create a new `AuthService`
+    #[inline]
+    pub fn new(inner: S, token: Option<Arc<HeaderValue>>) -> Self {
+        Self { inner, token }
+    }
+}
+
+impl<S, Body, Response> Service<Request<Body>> for AuthService<S>
+where
+    S: Service<Request<Body>, Response = Response>,
+{
+    type Response = S::Response;
+    type Error = S::Error;
+    type Future = S::Future;
+
+    #[inline]
+    fn poll_ready(&mut self, cx: &mut Context<'_>) -> Poll<std::result::Result<(), Self::Error>> {
+        self.inner.poll_ready(cx)
+    }
+
+    #[inline]
+    fn call(&mut self, mut request: Request<Body>) -> Self::Future {
+        if let Some(token) = self.token.as_ref() {
+            let _ = request
+                .headers_mut()
+                .insert(AUTHORIZATION, token.as_ref().clone());
+        }
+
+        self.inner.call(request)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[tokio::test]
+    async fn connect() {
```

Comment:
Why do we need such a trivial test here? It looks useless.

Reply:
Just test basic functionality creating a client to satisfy the ci coverage. I could remove this if not needed.

Reply:
done